lib/bolt/node_uri.rb

```diff
@@ -2,19 +2,20 @@
 
 module Bolt
   class NodeURI
-    def initialize(string)
-      @uri = parse(string)
+    def initialize(string, transport = 'ssh')
+      @uri = parse(string, transport)
     end
 
-    def parse(string)
+    def parse(string, transport)
```

Comment:
Not sure how much you care, but this method now allows for calling `parse` with a `transport` that is not `ssh|winrm|pcp`

Prior behavior wasn't really better though, because if you passed something like `http://foo.bar.com` you'd end up parsing `ssh://http://foo.bar.com`

Reply:
Let me fix that. We shouldn't be hardcoding valid transports here, we just need to know if a scheme was specified or not.

Reply:
Updated